numpy/lib/tests/test_function_base.py

```diff
@@ -1267,6 +1267,14 @@ def test_empty(self):
         assert_array_equal(a, np.array([0]))
         assert_array_equal(b, np.array([0, 1]))
 
+    def test_finite_range(self):
+        # Normal ranges should be fine
+        vals = np.linspace(0.0, 1.0, num=100)
+        histogram(vals, range=[0.25,0.75])
+        # non-finite values should raise AttributeError
+        assert_raises(ValueError, histogram, vals, range=[np.nan,0.75])
+        assert_raises(ValueError, histogram, vals, range=[np.inf,0.75])
```

Comment:
Might be a good idea to have non-finite values in the second entry of the `range` in some of the tests, in case a future change misses one of the entries.